architecture.md

```diff
@@ -0,0 +1,78 @@
+# Ignition Gazebo Architecture
+
+Ignition Gazebo is the application entry-point of Ignition architecture, generally encompassing the use of all other Ignition libraries.
+It is a library itself, but its use case described in this document is as an executable.
+As an executable, it runs a simulation by launching two processes: the backend server process and frontend client process.
+
+The core of Ignition Gazebo is the server-client communication and the various plugins it loads to facilitate the simulation.
+Plugins are loaded at runtime to create a link between other Ignition libraries and Ignition Gazebo so they can exchange information.
+Users can write their own plugins.
+Because they’re loaded at runtime, Ignition does not need to be recompiled to add or remove plugins.
+There are other types of plugins in the scope of Ignition; however, this use case only concerns Ignition Gazebo server and GUI plugins.
+
+Without plugins, Ignition libraries are not “aware” of each other and don’t possess the ability to interact.
+For example, Ignition Physics is independent of Ignition Gazebo, and vice versa.
+So Ignition Gazebo has a physics plugin that uses Ignition Physics as a library and incorporates Gazebo specifics, to enable Physics to be a system running in the simulation loop.
+This is the case for all libraries, whether their corresponding plugins affect the frontend or backend processes.
+
+The Ignition Physics library is only used in the physics plugin, not in other plugins nor in the core of Ignition Gazebo.
+Some libraries only have one corresponding plugin, or one in each process (frontend and backend).
+However, some libraries are used by all plugins, like Ignition Common which provides common functionality like logging, manipulating strings, file system interaction, etc., to all plugins.
+Other such libraries include Ignition Plugin, Ignition Math, SDFormat, and more.
```

Comment:
Would be nice to have links for each of these libraries mentioned, static ones independent of version numbers like https://ignitionrobotics.org/libs/math , but not essential.